python/hsfs/feature_store.py

```diff
@@ -534,6 +545,93 @@ def create_expectation(
             featurestore_id=self._id,
         )
 
+    def create_transformation_function(
+        self,
+        transformation_function: callable,
+        output_type: Union[
+            str,
+            TypeVar("str"),  # noqa: F821
+            TypeVar("string"),  # noqa: F821
+            bytes,
+            numpy.int8,
+            TypeVar("int8"),  # noqa: F821
+            TypeVar("byte"),  # noqa: F821
+            numpy.int16,
+            TypeVar("int16"),  # noqa: F821
+            TypeVar("short"),  # noqa: F821
+            int,
+            TypeVar("int"),  # noqa: F821
+            numpy.int,
+            numpy.int32,
+            numpy.int64,
+            TypeVar("int64"),  # noqa: F821
+            TypeVar("long"),  # noqa: F821
+            TypeVar("bigint"),  # noqa: F821
+            float,
+            TypeVar("float"),  # noqa: F821
+            numpy.float,
+            numpy.float64,
+            TypeVar("float64"),  # noqa: F821
+            TypeVar("double"),  # noqa: F821
+            datetime.datetime,
+            numpy.datetime64,
+            datetime.date,
+            bool,
+            TypeVar("boolean"),  # noqa: F821
+            TypeVar("bool"),  # noqa: F821
+            numpy.bool,
```

Comment:
Ok, but we also need to make sure this validation happens in the backend as well. Otherwise users will be able to create non-valid transformations from the UI